dandiapi/api/models/upload.py

```diff
@@ -126,6 +135,17 @@ class EmbargoedUpload(BaseUpload):
         Dandiset, related_name='embargoed_uploads', on_delete=models.CASCADE
     )
 
+    @classmethod
+    def object_key(cls, upload_id, **kwargs):
+        upload_id = str(upload_id)
+        dandiset_identifier = kwargs.get('dandiset')
+        dandiset_prefix = f'{dandiset_identifier}/' if dandiset_identifier is not None else ''
```

Comment:
Isn't a `dandiset_identifier` **required** for an `EmbargoedUpload`?

https://github.com/dandi/dandi-api/blob/embargo-design-doc/doc/design/embargo-mvp.md#models

Reply:
Good point.
I think I'm actually going to make `dandiset` a field on `Upload` as well as `EmbargoedUpload`, since as of https://github.com/dandi/dandi-api/pull/647/commits/3bbc1e871103d2130dff798d66798c58058cc06f it is required on all upload requests. e might as well track the dandiset for all uploads since we have the information.

Reply:
As asked elsewhere - if that would render already released/used dandi clients unable to upload to noon embargoed datasets, then let's not break API here

Reply:
That seems to me to be contradictory to https://github.com/dandi/dandi-api/issues/628.

Also, if we were to do that, what would happen if a user with an old CLI version naively tried to upload data to an embargoed dandiset? The upload would create a valid public AssetBlob, since the API has no way of knowing that the client intends for it to be embargoed. The CLI would then try to add that AssetBlob to the embargoed dandiset, which would also succeed, because public data is allowed to be included in embargoed dandisets. The result would be all the data uploaded publicly to the embargoed dandiset.

ping @yarikoptic @satra @waxlamp

Reply:
@yarikoptic - i think this simple change would also ensure that no person could upload a file without creating a dandiset first and having ownership of this dandiset (@dchiquito - is that checked). 

i think we should introduce this breaking change. i would be fine with an interim process, which suspends this requirement for non-embargoed dandisets, but is there really a reason why we could not deprecate former clients?

Reply:
i think @dchiquito comment is even more relevant as to why this should be introduced.

Reply:
(ownership is required to upload)

Reply:
yes, @dchiquito 's example is a good one.
But then how such migration would be implemented (besides clicking "release" simultaneously and somehow enforcing)?
- would current dandi-api (didn't try) allow (just ignore) `dandiset` parameter and not error out?
- if `yes` (no error) -- we should rush out dandi-cli release which would provide dandiset id, so whenever this is released, already released/installed client would be ready

Reply:
The server will currently ignore the `dandiset` parameter, so the CLI should be released first.

Reply:
ok, @jwodder already furnished a PR https://github.com/dandi/dandi-cli/pull/863 .  The only question is to either declare all prior versions "not good" or wait a little more (e.g. to may be discover other gotchas needing similar action) if deployment of embargo support is not imminent yet.